src/info/churn.rs

```diff
@@ -0,0 +1,155 @@
+        let file_churns = commit_metrics.file_churns_to_display.clone();
+        Self {
+            file_churns,
+            info_color,
+        }
+    }
+}
+impl std::fmt::Display for ChurnInfo {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        let mut churn_info = String::new();
+
+        let pad = self.title().len() + 2;
+
+        for (i, churn) in self.file_churns.iter().enumerate() {
+            let churn_str = churn.color(self.info_color);
+
+            if i == 0 {
+                let _ = write!(churn_info, "{churn_str}");
+            } else {
+                let _ = write!(churn_info, "\n{:<width$}{}", "", churn_str, width = pad);
+            }
+        }
+
+        write!(f, "{churn_info}")
+    }
+}
+
+#[typetag::serialize]
+impl InfoField for ChurnInfo {
+    fn value(&self) -> String {
+        self.to_string()
+    }
+
+    fn title(&self) -> String {
+        "Churn".into()
+    }
+
+    fn should_color(&self) -> bool {
+        false
+    }
+}
+
+fn truncate_file_path(path: &str, depth: usize) -> String {
+    let components: Vec<&str> = path.split('/').collect();
+
+    if depth == 0 || components.len() <= depth {
+        return path.to_string();
+    }
+
+    let truncated_components: Vec<&str> = components
+        .iter()
+        .skip(components.len() - depth)
+        .copied()
+        .collect();
+    let truncated_path = format!(".../{}", truncated_components.join("/"));
+
+    truncated_path
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_display_file_churn() {
+        let file_churn = FileChurn::new("path/to/file.txt".into(), 50, NumberSeparator::Plain);
+
+        assert_eq!(file_churn.to_string(), ".../to/file.txt 50");
+    }
+
+    #[test]
+    fn test_churn_info_value_with_two_file_churns() {
+        let file_churn_1 = FileChurn::new("path/to/file.txt".into(), 50, NumberSeparator::Plain);
+        let file_churn_2 = FileChurn::new("file_2.txt".into(), 30, NumberSeparator::Plain);
+
+        let churn_info = ChurnInfo {
+            info_color: DynColors::Rgb(255, 0, 0),
+            file_churns: vec![file_churn_1, file_churn_2],
+        };
+
+        assert!(churn_info.value().contains(
+            &".../to/file.txt 50"
+                .color(DynColors::Rgb(255, 0, 0))
+                .to_string()
+        ));
+
+        assert!(churn_info
+            .value()
+            .contains(&"file_2.txt 30".color(DynColors::Rgb(255, 0, 0)).to_string()));
+    }
+
+    #[test]
+    fn test_truncate_file_path() {
+        assert_eq!(
+            truncate_file_path("path/to/file.txt", 3),
+            "path/to/file.txt"
+        );
+        assert_eq!(
+            truncate_file_path("another/file.txt", 2),
+            "another/file.txt"
+        );
+        assert_eq!(truncate_file_path("file.txt", 1), "file.txt");
+        assert_eq!(truncate_file_path("path/to/file.txt", 2), ".../to/file.txt");
```

Comment:
I wonder if an actual ellipsis could be used here, as it's smaller usually which I prefer visually: `...` `…`

Reply:
Great suggestion, done.